compiler/rustc_error_codes/src/error_codes/E0446.md

```diff
@@ -1,16 +1,16 @@
-A private type was used in a public type signature.
+A private type or trait was used in a public associated type signature.
 
 Erroneous code example:
 
 ```compile_fail,E0446
-#![deny(private_in_public)]
-struct Bar(u32);
-
-mod foo {
-    use crate::Bar;
-    pub fn bar() -> Bar { // error: private type in public interface
-        Bar(0)
-    }
+struct Bar;
```

Comment:
would the above `compile_fail,E0446` need to be modified at some point too? Probably needs a `<!-- FIXME: ... -->` of some sort?